project_event/models/project_task.py

```diff
@@ -282,12 +282,43 @@ def create_new_activity(self, vals):
             .next_by_code('project.task.activity')
         return super(Task, self).create(vals)
 
+    @api.multi
+    def update_reservation_event(self, update_task):
+        if update_task and self.reservation_event_id:
+            self.ensure_one()
+            reservation_event = self.env['calendar.event'].browse(
+                self.reservation_event_id)
+            partners = []
+            for e in self.employee_ids:
+                if e.user_id:
+                    partners.append(e.user_id.partner_id.id)
+                else:
+                    raise UserError(
+                        _('Please define user account for the %s employee') % (
+                            e.name,))
+            reservation_event.write(
+                {
+                    'start': self.date_start,
+                    'stop': self.date_end,
+                    'name': self.name,
```

Comment:
only start and stop have different keys (vs date_start and date_end), for the others you could pop these  if present and then dictionary.update(vals) and then write